docs/locale/ja/Architecture.md

```diff
@@ -0,0 +1,353 @@
+
+```
+persistent://my-property/us-w/my-app1/my-topic-1
+persistent://my-property/us-w/my-app1/my-topic-2
+...
+```
+
+
+### Producer
+
+Producerはトピックに接続し、メッセージを送信します。
+
+**同期送信 vs. 非同期送信** - メッセージは同期的あるいは非同期的にBrokerに送信できます:
+
+- 同期: Producerはメッセージを送信した後Brokerからの確認応答を待ちます。
+- 非同期: Producerはメッセージをブロッキングキューに入れ、制御を戻します。クライアントライブラリはバックグラウンドでBrokerに送信します。キューが最大量 (設定可能) に達した場合、Producerは送信APIを呼び出した時、Producerの引数に応じてブロックされる、あるいは失敗する可能性があります。
+
+**圧縮** - 帯域を節約するため、メッセージを圧縮することが可能です。圧縮と解凍はどちらもクライアントで動作します。以下の圧縮形式をサポートしています:
+
+- LZ4
+- ZLIB
+
+**バッチ** - バッチ処理が可能な場合、Producerはメッセージを蓄積し、1つのリクエストでメッセージのバッチを送信しようとします。バッチサイズはメッセージの最大数と最大発行レイテンシで定義されます。
+
+### Consumer
+
+Consumerはサブスクリプションに接続し、メッセージを受け取ります。
+
+**同期受信 vs. 非同期受信** - 同期受信はメッセージが利用可能になるまでブロックされます。同期受信ではCompletableFutureインスタンスとしてすぐに返却されます。CompletableFutureインスタンスは新しいメッセージが利用可能になった時、受信して完了します。
+
+**確認応答** - メッセージは1つ1つ個別に、あるいは累積的に確認応答が返されます。累積的な確認応答をするConsumerは、最後に受け取ったメッセージの確認応答を返します。その場合、確認応答を返したメッセージまでのストリーム内の全てのメッセージはそのConsumerに再送されません。サブスクリプションモードがSharedの場合は、累積的な確認応答は使用できません。
+
+**リスナー** - カスタマイズされたMessageListenerの実装をConsumerに渡すことができます。クライアントライブラリは新しいメッセージを受け取るとリスナーを呼び出します (Consumer Receiveを呼び出す必要はありません) 。
+
+
+### パーティションドトピック
+
+通常のトピックは最大スループットを制限する1つのBrokerのみから提供されます。パーティションドトピックは高いスループットを達成するために複数のBrokerにまたがる特殊なトピックです。パーティションドトピックは管理API/CLIを通じて明示的に作成される必要があります。トピックを作成する際、パーティションの数を記述できます。
+
+パーティションドトピックは実際にはN (パーティションの数) 個の内部トピックとして実装されます。内部トピックと普通のトピックについて、サブスクリプションモードの挙動に違いはありません。
+
+![パーティションドトピック](../../img/pulsar_partitioned_topic.jpg)
+
+**ルーティングモード** - ルーティングモードはどのパーティション (内部トピック) にメッセージが発行されるかを決定します:
+
+- キーハッシュ: メッセージにキーが指定されていれば、Partitioned Producerはキーをハッシュ化し、キーバケットごとの順序を保証しながら、特定のパーティションに割り当てます。
+- Single Partition: キーが指定されない場合、各プロデューサのメッセージはプロデューサごとのメッセージの順序を保証するために、 (最初にランダムに選ばれた) 専用パーティションにルーティングされます。
+- Round Robin Partition: キーが指定されない場合、最大スループットを達成するために、全てのメッセージはラウンドロビンで異なるパーティションにルーティングされます。順序は保証されません。
+- Custom Partition: メッセージは独自のMessageRouterの実装によってルーティングされます。
+
+```java
+public interface MessageRouter extends Serializable {
+    /**
+     * @param msg メッセージオブジェクト
+     * @return そのメッセージのために使用するパーティションのインデックス
+     */
+    int choosePartition(Message msg);
+}
+```
+
+### 永続化
+
+保証されたメッセージの配送はそれらが配信され、Consumerによる確認応答を受け取るまで頑健な方法で保存される必要があります。このメッセージングの方法は一般的にPersistent Messagingと呼ばれます。
+
+メッセージの永続性はトピックレベルで設定されます。トピックは“persistent”にも“non-persistent”にもなりえます。トピック名は以下のようになります:
+*persistent://my-property/global/my-ns/my-topic*
+
+**persistent**
+
+- 全てのメッセージはディスク上に保存、同期され、N個のコピー (例: 4個のコピーが個別にミラーリングされたRAIDボリュームを持つ2台のサーバに存在) が全てのConsumerがメッセージをconsumeするまで保存されます。カーソルと呼ばれるCosumerの購読位置もディスク上に保存されます。メッセージは必ず1回は配送されることが保証されます。
+
+**non-persistent**
+
+- 現在、PulsarはNon-persistent Messagingとして知られるベストエフォートの配送はサポートしていません。しかし、今後の計画としてNon-persistent Messagingをサポートする予定があります。
+
+
+### レプリケーション
+
+Pulsarでは異なるジオロケーションでのメッセージのproduce, consumeが可能です。例えば、アプリケーションが1つのジオ/マーケットにデータを発行し、異なるジオ/マーケットでそれをconsumeし処理したいとします。Global Replicationがそれを可能にします。
+
+### 認証と認可
+
+TODO
```

Comment:
Since it has already been added at the latest,
we will also update here next time when we translate v1.17